### PR TITLE
(PXP-6359) Add search filter to explorer

### PR DESCRIPTION
**Pull request by em-ingram**
Jira Ticket: [PXP-6359](https://ctds-planx.atlassian.net/browse/PXP-6359)

### New Features
- Adds a new filter type, `searchFilter`, to the available filters. Search Filters allow the user to search over fields which have too many options to be displayed normally -- for example, individual subject IDs or file GUIDs.

### Dependency updates
- Add `react-select` and `react-select-async-paginate` dependencies

### CODE REVIEW

File: src/components/filters/FilterSection/index.jsx [hunk 1]
```diff
@@ -297,37 +326,71 @@ class FilterSection extends React.Component {
...
+        }
         {
           isTextFilter && this.getSearchInput()
         }
         <div className='g3-filter-section__options'>
           {
-            this.state.isExpanded
+            isSearchFilter && this.state.isExpanded
+              ? this.props.options
+                // .filter(option => this.state.optionsVisibleStatus[option.text])
```
Comment (mfshao) [inline src/components/filters/FilterSection/index.jsx]: is this needed?

File: src/components/filters/FilterSection/index.jsx [hunk 2]
```diff
@@ -297,37 +326,71 @@ class FilterSection extends React.Component {
...
-                        count={option.count}
-                        hideZero={this.props.hideZero}
-                        accessible={option.accessible}
-                        tierAccessLimit={this.props.tierAccessLimit}
-                        disabled={option.disabled}
-                        lockedTooltipMessage={this.props.lockedTooltipMessage}
-                        disabledTooltipMessage={this.props.disabledTooltipMessage}
-                      />
-                    );
+                  return (
```
Comment (mfshao) [inline src/components/filters/FilterSection/index.jsx:353]: feels like this part is a bit redundant comparing to https://github.com/uc-cdis/gen3-ui-component/pull/78/files#diff-07a74b594d96598ad424ba2c977d9a88R337-R357

maybe we can filter the `options` according to the controlling variables before?

**Review by mfshao — COMMENTED** [review]
looks cool, a couple of minor comments

**Review by ZakirG — APPROVED** [review]
very cool. i like how the selected options remain highlighted in the select dropdown ~

**Comment by ZakirG** [comment]
something i noticed in the stories view: searching "99" will bring up "guid-99", but searching "guid" won't bring up any options, nor will searching "guid-99". is that intended behavior?

File: src/components/filters/FilterSection/index.jsx (hunk 1 above)
Comment (em-ingram) [reply src/components/filters/FilterSection/index.jsx]: No! Woops, I'll delete that

File: src/components/filters/FilterSection/index.jsx (hunk 2 above)
Comment (em-ingram) [reply src/components/filters/FilterSection/index.jsx:353]: Gotcha, let me try to un-duplicate this code -- I'll see if I can do it in a way that isn't too confusing.

**Comment by em-ingram** [comment]
> something i noticed in the stories view: searching "99" will bring up "guid-99", but searching "guid" won't bring up any options, nor will searching "guid-99". is that intended behavior?

Nice catch, that's a bug in the quick mock search function I wrote for the filters. https://github.com/uc-cdis/gen3-ui-component/blob/923e18527c1ee1f23f7196d67916eb05c462c068/stories/filters.js#L209 Should be `>= 0` or `!== -1`. Fixed!

**Comment by ZakirG** [comment]
> > something i noticed in the stories view: searching "99" will bring up "guid-99", but searching "guid" won't bring up any options, nor will searching "guid-99". is that intended behavior?
> 
> Nice catch, that's a bug in the quick mock search function I wrote for the filters.
> 
> https://github.com/uc-cdis/gen3-ui-component/blob/923e18527c1ee1f23f7196d67916eb05c462c068/stories/filters.js#L209
> 
> Should be `>= 0` or `!== -1`. Fixed!

it works thanks ~

**Review by mfshao — APPROVED** [review]
👍

**Comment by mfshao** [comment]
oh don't forget to update the version number

**Comment by em-ingram** [comment]
> oh don't forget to update the version number 

thank you, good catch!

**Pull request merged**